demo/src/screens/ThankScreen/ThankScreen.js

```diff
@@ -1,13 +1,88 @@
-import React from 'react';
+import React, {useMemo} from 'react';
+import {Container, Row, Col, Jumbotron} from 'react-bootstrap';
+import {ProductListing} from '../../components/ProductListing/ProductListing.js';
+import {connect} from 'react-redux';
+import PropTypes from 'prop-types';
 
 /**
+ * @param {!Object} items The items stored in the site state
  * @return {!JSX} Page component where we thank the user
  *     for their purchase(s).
  */
-export function ThankScreen() {
+export const ThankScreenBase = function({items}) {
+  const /** Array<!JSX> */ productListings = useMemo(() => {
+    const ITEMS_TO_DISPLAY = 6;
+    const products = [];
+    const usedItemSet = new Set();
+
+    let i;
+    for (i = 0; i < ITEMS_TO_DISPLAY; ++i) {
+      let maxPrice = 0;
+      let maxPriceItem;
+      let maxPriceID;
+
+      // Finds and pushes the most expensive item that has not yet been used.
+      for (const [itemID, item] of Object.entries(items)) {
+        if (item.cost > maxPrice && !usedItemSet.has(itemID)) {
+          maxPriceItem = item;
+          maxPriceID = itemID;
+          maxPrice = item.cost;
+        }
+      }
+      products.push(
+          <li className="product-listing-container" key={maxPriceID}>
+            <ProductListing
+              id={maxPriceID}
+              title={maxPriceItem.name}
+              price={maxPriceItem.cost}
+              imageSrc={maxPriceItem.thumbnail}
+              description={maxPriceItem.description}
+            />
+          </li>,
+      );
+      usedItemSet.add(maxPriceID);
+    }
+
+    return products;
+  }, [items]);
+
   return (
-    <div>
-      <h1>{`I'm the Thanks Page!`}</h1>
-    </div>
+    <Container>
+      <Row>
+        <Col/>
+        <Col xs={10}>
+          <Jumbotron>
+            <div style = {{paddingBottom: 50, fontSize: 30}}>
+              {'Thank you for your purchase!'}
+            </div>
+            {'Here are some other items you may be interested in:'}
+            <br/>
+            <ul className="product-list">
+              {productListings}
+            </ul>
+          </Jumbotron>
+        </Col>
+        <Col/>
+      </Row>
+    </Container>
   );
-}
+};
+
+ThankScreenBase.propTypes = {
+  items: PropTypes.object,
+};
+
+/**
+  * @param {{items:Itemstore}} state
+  * @return {{items:Itemstore}} The items stored in the site state.
+  */
+const mapStateToProps = (state) => ({
```

Comment:
I'm confused about what this method does? How is it different from just returning the state as-is or `Object.assign({}, state);` if the intention is to make a copy.

Reply:
I am also still a bit confused by this, but The react-redux connect function takes a mapStateToProps function so that it can merge the selected parts of the redux state with the component's props

Reply:
Won't suggested `Object.assign({}, state)` solve this problem? (the advantage is we reduce the maintenance burden of cherry-picking properties)

Reply:
Ok so I talked to KJ since she has worked with react quite a bit. We went over the react redux documentation and came up with a conclusion for why mapStateToProps is being used this way. We aren't actually making a copy, we are cherry picking properties of state because mapStateToProps passes the property as a React prop to the ThanksScreenBase component. If we pass in the entire state, the component will re-render for every state change instead of only re-rendering when the items object is changed. Since this component is only using a a specific part of the state, it will only re-render when that part changes. Here is the documentation we used https://react-redux.js.org/using-react-redux/connect-mapstate#return !

Reply:
Ah, this makes sense now.
Thank you for looking into this!